src/main.cc

```diff
@@ -763,11 +765,23 @@ master_revive_kids(int sig)
 #endif
 }
 
+void
+master_restart(int sig)
+{
+    RestartSignal = sig;
+    do_restart = 1;
+
+#if !_SQUID_WINDOWS_
+#if !HAVE_SIGACTION
+    signal(sig, master_revive_kids);
+#endif
+#endif
+}
+
 /// Shutdown signal handler for master process
 void
 master_shutdown(int sig)
 {
-    do_shutdown = 1;
```

Comment:
Why should this assignment be removed? Did its presence cause any bugs other than the SIGTTIN handling bug?